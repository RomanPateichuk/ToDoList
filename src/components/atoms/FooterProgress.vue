<template>
  <div :class="$style.progress">{{ X }}/{{ getActiveCount }} left</div>
</template>

<script>
import { mapGetters } from "vuex";

export default {
  computed: {
    ...mapGetters(["getActiveCount"]),

    // getTaskActiveCount: function () {
    //   return this.get_tasks_active.reduce(
    //     (prev, el) => (el.checked == false ? prev + 1 : prev),
    //     0
    //   );
    // },
    // getTaskDoneCount: function () {
    //   return this.get_tasks_active.reduce(
    //     (prev, el) => (el.checked == true ? prev + 1 : prev),
    //     0
    //   );
    // },
  },
  methods: {
    // getTaskDoneCount: function(){
    //   return this.get_tasks_active.reduce((prev, el) => (el.checked == true) ? prev + 1: prev, 0);
    // },
  },
};
</script>

<style lang="scss" module>
@import "@/assets/scss/main.scss";
</style>
