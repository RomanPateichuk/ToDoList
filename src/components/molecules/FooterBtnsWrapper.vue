<template>
  <div :class="$style.btn">
    <FooterBtn
      v-for="(btn, index) in btns"
      :key="index"
      :FooterBtnValue="btn"
      @callFilter="callMuttationSetFilter(btn)"
    />
  </div>
</template>
<script>
export default {
  components: {
    FooterBtn: () => import("@/components/atoms/FooterBtn.vue"),
  },
  data() {
    return {
      btns: ["All", "Active", "Completed"],
    };
  },
  methods: {
    callMuttationSetFilter: function (FooterBtnValue) {
      this.$store.commit("setFilter", FooterBtnValue);
    },
  },
};
</script>
<style lang="scss" module>
@import "@/assets/scss/main.scss";
.btn {
  :not(:last-child) {
    margin-right: 1rem;
    @include MediaFooterBntmobile;
  }
}
</style>
