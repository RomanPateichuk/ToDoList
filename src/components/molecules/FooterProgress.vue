<template>
  <div :class="$style.progress">1/3 left</div>
</template>

<style lang="scss" module>
@import "@/assets/scss/main.scss";
</style>
