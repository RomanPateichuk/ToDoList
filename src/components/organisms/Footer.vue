<template>
  <footer class="footer">
    <FooterProgress />
    <FooterBtnWrapper />
  </footer>
</template>

<script>
export default {
  components: {
    FooterBtnWrapper: () => import("@/components/atoms/FooterBtnWrapper.vue"),
    FooterProgress: () => import("@/components/molecules/FooterProgress.vue"),
  },
};
</script>

<style scoped lang="scss">
@import "@/assets/scss/main.scss";
.footer {
  display: flex;
  min-height: 3.125rem;
  justify-content: space-between;
  align-items: center;
  background: $task-remove;
  padding: 0 1.875rem;
  font-family: "Open Sans", sans-serif;
  @media (max-width: 410px) {
    font-size: 1rem;
    padding: 0 1rem;
  }
  @media (max-width: 314px) {
    flex-direction: column;
  }

  * {
    color: $footer-font-color;
  }
}
</style>
