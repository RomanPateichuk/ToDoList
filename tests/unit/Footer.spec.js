import { createLocalVue, shallowMount } from "@vue/test-utils";
import Footer from "../../src/components/organisms/Footer.vue";
import FooterProgress from "../../src/components/atoms/FooterProgress.vue";
import FooterBtnsWrapper from "../../src/components/molecules/FooterBtnsWrapper.vue";
import Vuex from "vuex";

const localVue = createLocalVue();
localVue.use(Vuex);

const getters = {
  getDoneCount: () => 2,
  getTasks: () => [{ id: "1", value: "Task 1", checked: false }],
};

const store = new Vuex.Store({ getters });

describe("Тестирование компонента Footer:", () => {
  const wrapper = shallowMount(Footer, { store });
  test("соответствие снимку", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("Подключение компонента 'FooterProgress'", () => {
    expect(wrapper.findComponent(FooterProgress).exists()).toBe(true);
  });

  test("Подключение компонента 'FooterBtnsWrapper'", () => {
    expect(wrapper.findComponent(FooterBtnsWrapper).exists()).toBe(true);
  });
});
