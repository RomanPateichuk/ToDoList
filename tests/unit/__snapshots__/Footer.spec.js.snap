// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Тестирование компонента Footer: соответствие снимку 1`] = `
<footer>
  <!---->
  <div><label><input type="radio" name="radio-footer"><span>All</span></label><label><input type="radio" name="radio-footer"><span>Active</span></label><label><input type="radio" name="radio-footer"><span>Completed</span></label></div>
</footer>
`;
